pkg/operator/ceph/file/filesystem.go

```diff
@@ -160,13 +156,35 @@ func newFS(name string, metadataPool *model.Pool, dataPools []*model.Pool, activ
 	}
 }
 
+// SetPoolSize method sets the sizes for MetadataPool and dataPool
+func SetPoolSize(context *clusterd.Context, clusterName, fsName string, dataPool []*model.Pool, metadataPool *model.Pool) error {
+	// updating the MetaDataPool size
+	metadataPool = GenerateMetadataPoolName(fsName, metadataPool)
+	newPool := client.ModelPoolToCephPool(*metadataPool)
```

Comment:
thanks @travisn I made some changes but still not sure if they contain all the cases that we need.
I'm still confused if I call the full `CreatePoolWithProfile()` method  instead of `SetPoolSize()` OR
 implement `SetPoolSize()` method in the similar pattern as `CreatePoolWithProfile()`
Please review it and let me know what am I missing.
Thanks

Reply:
You should be able to call `CreatePoolWithProfile()` once for the metadata pool and once for each of the data pools. That method will handle everything about replicated or EC settings for you. But it will be very important to test the behavior. For example, please verify:
- You can create an EC pool.
- You can create a replicated pool with replication 1, then change the size to 2 (or something else).
- Restart the operator and verify that the operator still applied the latest settings, and did not hit any errors when it processed the pools.